File: camping_angular/src/app/service/randonner.service.ts
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
const AUTH_API = 'http://localhost:8080/rand';
const httpOptions = {
  headers: new HttpHeaders({ 'Content-Type': 'application/json' })
};
@Injectable({
  providedIn: 'root'
})
export class RandonnerService {
  constructor(private http: HttpClient) { }

  register(rand:any): Observable<any> {
    return this.http.post(AUTH_API + '/add', {
      adress:rand.adress,
      programme:rand.programme,
      cercuit:rand.cercuit,
      difficult:rand.difficult,
      date:rand.date,
      imageUrl:rand.imageUrl,
    }, httpOptions);
  }


 
}

File: campingomobile/lib/services/RandonnerService.dart
import 'dart:convert';

import 'package:Campingo/constants/constants.dart';
import 'package:http/http.dart'as http;
addrandonner (
      String adress,
      String programme,
      String cercuit,
      String difficult,
      String date,
      String imageUrl,
      // String imageUrl
      ) async {
    var response = await http.post(Uri.parse('$BaseUrl$AddRand/$id'),
        headers: <String, String>{"Content-Type": "application/json"},
        body: jsonEncode(<String, String>{
          "adress": adress,
          "programme": programme,
          "cercuit": cercuit,
          "difficult": difficult,
          "date": date,
          // "image": imageUrl,
        }));

    String responseString = response.body;
  }